### DISCUSSION TITLE
Thank you!

### DISCUSSION THREAD

**Discussion by nzRegularIT** [vmware/powershell-module-for-vmware-cloud-foundation-password-management#60] [Community Help] [answered]
Hey team,

This is a great set of modules, I am looking at using them across multiple VCF deployments to set and maintain password policies. I am currently testing this in UAT on both Windows, using Windows PowerShell and PowerShell Core, and also on Linux using PowerShell Core.

One thing I noticed was the use of both `Debug-CatchWriter` and `Debug-ExceptionWriter`, from different modules, `PowerVCF` and `PowerValidatedSolutions`. I was just curious as to the difference between the two?

**Comment by tenthirtyam** [comment, marked as answer]
Hi Aaron! 👋 

Thanks for your comment on these open-source modules - it's much appreciated!

You'll find that the `Debug-CatchWriter` from `PowerVCF` is used in the `Publish-`, `Invoke-`, and `Start-` functions as those write to a log file using `Write-LogMessage` from `PowerVCF`. 

All other cmdlets use `Debug-ExceptionWriter` to output to the console.

That said, I do see some technical debt that we will address in other modules to ensure consistency.

Ryan

cc @sowjuec
